gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -394,8 +394,8 @@ private Map<String, AddSpecResponse> updateOrAddSpecHelper(Spec spec, boolean tr
       if (!response.getValue().getFailures().isEmpty()) {
         for (Map.Entry<SpecCatalogListener, CallbackResult<AddSpecResponse>> entry : response.getValue().getFailures().entrySet()) {
           Throwable error = entry.getValue().getError();
-          if (error instanceof TooSoonToRerunSameFlowException) {
-            throw (TooSoonToRerunSameFlowException) error;
+          if (error instanceof RuntimeException && error.getCause() instanceof TooSoonToRerunSameFlowException) {
+            throw (TooSoonToRerunSameFlowException) error.getCause();
```

Comment:
the cast isn't necessary (the reason I suggested wrapping `TooSoonToR...` was to enable uniform, type-agnostic code here.)

for a method with the signature `throws Throwable`, aren't these two equivalent?
```
throw error.getCause();
throw (T) error.getCause();
```

Reply:
makes sense updated